Libraries/VirtualFunctions/VirtualFunctions.h

```diff
@@ -0,0 +1,312 @@
+        char_map[row_char][column_char + 1] = '.';
+      }
+      else {
+        char_map[row_char][column_char + 1] = 'W';
+      }
+      if(tiles_map.getTile(row_tile, column_tile).ableToGoSouth()) {
+        char_map[row_char + 1][column_char] = '.';
+      }
+      else {
+        char_map[row_char+1][column_char] = 'W';
+      }
+      if(tiles_map.getTile(row_tile, column_tile).ableToGoWest()) {
+        char_map[row_char][column_char - 1] = '.';
+      }
+      else {
+        char_map[row_char][column_char - 1] = 'W';
+      }
+      column_char += 2;
+    }
+    row_char += 2;
+  }
+
+  char_map[tiles_map.currentRow()*2 + 1][tiles_map.currentColumn()*2 + 1] = 'R';
+  
+  for(uint8_t i = 0 ; i < tiles_map.numberOfRows()*2 + 1 ; ++i) {
+    for(uint8_t j = 0 ; j < tiles_map.numberOfColumns()*2 + 1 ; ++j) {
+      Serial.print(char_map[i][j]);
+      Serial.print(" ");
+    }
+    Serial.println(" ");
+  }
+  Serial.println(" ");
+}
+
+// Function that moves the position of the robot in the virtual map according to a char vector recieved and returns the resulting tiles map.
+Map followPath(TVector<char> path, Map tiles_map) {
+  while (path.getSize() > 0) {
+    if (path[0] == 'N') {
+      tiles_map.moveNorth();
+      moveMapNorth();
+    }
+    else if (path[0] == 'E') {
+      tiles_map.moveEast();
+      moveMapEast();
+    }
+    else if (path[0] == 'S') {
+      tiles_map.moveSouth();
+      moveMapSouth();
+    }
+    else if (path[0] == 'W') {
+      tiles_map.moveWest();
+      moveMapWest();
+    }
+
+    Serial.println("|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||");
+    printVirtualMap();
+    Serial.println("");
+
+    path.popFirst();
+  }
+
+  return tiles_map;
+}
+
+// Function that updates the tiles map according to the virtual map and returns a tiles_map.
+Map updateTiles(Map tiles_map) {
+  Tile current_tile;
+
+  current_tile = tiles_map.currTile();
+    current_tile.setVisited();
+
+    if (virtual_map[tiles_map.currentRow()*2 + 1][tiles_map.currentColumn()*2 + 1] == '1') {
+      current_tile.setWeight(1);
+    }
+    else if (virtual_map[tiles_map.currentRow()*2 + 1][tiles_map.currentColumn()*2 + 1] == '2') {
+      current_tile.setWeight(2);
+    }
+    else if (virtual_map[tiles_map.currentRow()*2 + 1][tiles_map.currentColumn()*2 + 1] == '3') {
+      current_tile.setWeight(3);
+    }
+    
+    // Update North.
+    if (virtual_map[tiles_map.currentRow()*2][tiles_map.currentColumn()*2 + 1] != 'W') {
+      current_tile.setNorth();
+      if (tiles_map.currentRow() > 0 && tiles_map.northTile().isVisited() == false) {
+        Tile north_tile = tiles_map.northTile();
+        north_tile.setSouth();
+        north_tile.setAccessible();
+        tiles_map.setTile(north_tile, 'N');
+      }
+      else if (tiles_map.currentRow() == 0) {
+        Tile north_tile;
+        north_tile.setSouth();
+        north_tile.setAccessible();
+        tiles_map.addRowFirst(north_tile);
+      }
```

Comment:
Is it possible to make a general function of this? Because at first sight the following lines seems like a bunch of repeated code.

Reply:
I was working on that, but actually a lot of lines change. It conserves the same structure on each direction to update, but the functions are different on each case. I think it is better to leave it that way.

Reply:
Ok, we can further discuss this later, please add a TODO at the beginning of this section to revise it on a future PR.

Reply:
Ok